client/signup/config/flows.js

```diff
@@ -258,7 +265,24 @@ function filterDesignTypeInFlow( flow ) {
 
 function filterFlowName( flowName ) {
 	const defaultFlows = [ 'main', 'website' ];
-	// do nothing. No flows to filter at the moment.
+
+	/**
+	 * Only run the User First Signup for logged out users.
+	 */
+	if ( ! user.get() ) {
+		if ( includes( defaultFlows, flowName ) && abtest( 'userFirstSignup' ) === 'userFirst' ) {
+			return 'userfirst';
+		}
+
+		/**
+		 * Users should not be able to reach `userfirst-secondary` without being logged in, since
+		 * it doesn't contain the `user` step to register a user and site respectively.
+ 		 */
+		if ( flowName === 'userfirst-secondary' ) {
+			return 'main';
```

Comment:
Good catch. It was a leftover from a previous version. Updated to send the user to `userfirst`.